R/SEIaImIsRD.R

```diff
@@ -0,0 +1,352 @@
+    mu = list(i_asymptomatic = NA_real_, i_mild = NA_real_, i_severe = NA_real_))) {
+    param_list <- value
+    names(param_list) <- object@transmission_parameter_names
+    object@transmission_parameters <- param_list
+    # check if values are valid
+    errors <- character()
+    # check whether all required parameters are set
+    is_na_params <- is.na(param_list)
+    if (sum(is_na_params) != 0) {
+      msg <- paste("Missing parameters:",
+                   paste(names(param_list)[is_na_params], collapse = ", ")
+      )
+      errors <- c(errors, msg)
+    }
+    # check whether the lengths of beta, gamma and mu correspond to the number of infected groups
+    n_beta <- length(param_list$beta)
+    if (n_beta != 3) {
+      msg <- paste0(
+        "Length of parameter beta,", n_beta, ", is not equal to the setting ", 3)
+      errors <- c(errors, msg)
+    }
+    n_gamma <- length(param_list$gamma)
+    if (n_gamma != 3) {
+      msg <- paste0(
+        "Length of parameter gamma,", n_gamma, ", is not equal to the setting ", 3)
+      errors <- c(errors, msg)
+    }
+    n_mu <- length(param_list$mu)
+    if (n_mu != 3) {
+      msg <- paste0(
+        "Length of parameter mu,", n_mu, ", is not equal to the setting ", 3)
+      errors <- c(errors, msg)
+    }
+    n_p_symptom <- length(param_list$p_symptom)
+    if (n_p_symptom != 2) {
+      msg <- paste0(
+        "Length of parameter p_symptom,", n_mu, ", is not equal to the setting ", 2)
+      errors <- c(errors, msg)
+    }
+    else if (param_list$p_symptom %>% unlist %>% sum > 1){
+      msg <- paste0("Sum of p_symptom, ", param_list$p_symptom %>% unlist %>% sum, ", is greater than ", 1)
+      errors <- c(errors, msg)
+    }
+    if (length(errors) == 0) {
+      object@transmission_parameters <- param_list
+      object
+      }
+    else stop(paste(errors, ", please check and rerun transmission_parameters<-.\n"))
+    })
+
+#' Solves the ode system.
+#'
+#' \deqn{\frac{dS(t)}{dt} = -S(t)(\beta.i_{asymptomatic} I_{asymptomatic}(t) + \beta.i_{mild} I_{mild}(t) + \beta.i_{severe} I_{severe}(t)) + \omega R(t)}
+#' \deqn{\frac{dE(t)}{dt} = S(t)(\beta.i_{asymptomatic} I_{asymptomatic}(t) + \beta.i_{mild} I_{mild}(t) + \beta.i_{severe} I_{severe}(t)) - \kappa E(t)}
+#' \deqn{\frac{dI_{asymptomatic}(t)}{dt} = p_symptom.i_{asymptomatic} \kappa E(t) - (\gamma.i_{asymptomatic} + \mu.i_{asymptomatic}) I_{asymptomatic}(t)}
+#' \deqn{\frac{dI_{mild}(t)}{dt} = p_symptom.i_{mild}\kappa E(t) - (\gamma.i_{mild} + \mu.i_{mild}) I_{mild}(t)}
+#' \deqn{\frac{dI_{severe}(t)}{dt} = p_symptom.i_{severe}\kappa E(t) - (\gamma.i_{severe} + \mu.i_{severe}) I_{severe}(t)}
+#' \deqn{\frac{dR(t)}{dt} = -\omega R(t) + \gamma.i_{asymptomatic} I_{asymptomatic}(t) + \gamma.i_{mild} I_{mild}(t) + \gamma.i_{severe} I_{severe}(t)}
+#' \deqn{\frac{dD(t)}{dt} =  \mu.i_{asymptomatic} I_{asymptomatic}(t) + \mu.i_{mild} I_{mild}(t) + \mu.i_{severe} I_{severe}(t)}
```

Comment:
I'm not sure we need the double subscripts. How about just (say) \beta_{mild} opposed to \beta.i_{mild}? Similarly for all the parameters that vary according to symptom group.